File: Caladan.Frontend/ViewModels/Transaction.cs
using System;

namespace Caladan.Frontend.ViewModels
{
    public class Transaction : Base
    {
        public string TransactionHash { get; set; }
        public string OriginalTransactionHash { get; set; }
        public ulong BlockNumber { get; set; }
        public ulong TransactionIndex { get; set; }
        public string BlockHash { get; set; }
        public string From { get; set; }
        public string To { get; set; }
        public ulong Gas { get; set; }
        public ulong GasPrice { get; set; }
        public ulong GasUsed{ get; set; }
        public double Fee { get; set; }
        public double Value { get; set; }
        public string Symbol { get; set; }
        public string Input { get; set; }
        public ulong Nonce { get; set; }
        public ulong Timestamp { get; set; }
        public double PriceUsd { get; set; }
        public double PriceBtc { get; set; }
        public double PriceEur { get; set; }
        public string ConfirmedOnFormatted { get; set; }
        public ulong Confirmations { get; set; }
        public string Url => $"/transaction/{TransactionHash}";
        public TransactionReceipt Receipt { get; set; }
        public string ReceiptRaw { get; set; }
        public string Raw { get; set; }
    }
}

File: Caladan.Frontend/ClientApp/components/transaction/transaction.ts
import Vue from 'vue';
import { Component } from 'vue-property-decorator';

interface Transaction {
    TransactionHash: string;
    BlockNumber: number;
    TransactionIndex: number;
    BlockHash: string;
    From: string;
    To: string;
    Gas: number;
    GasPrice: number;
    GasUsed: number;
    Fee: number;
    Confirmations: number;
    Value: number;
    Symbol: string;
    Input: string;
    Nonce: number;
    Timestamp: number;
    PriceUsd: number;
    PriceBtc: number;
    PriceEur: number;
    ConfirmedOnFormatted: string;
    Found: boolean;
    Url: string;
    ReceiptRaw: string;
    Raw: string
}

@Component
export default class TransactionComponent extends Vue {
    transaction: Transaction;

    data() {
        return {
            transaction: null
        }
    }
    mounted() {
        this.loadData();
        setInterval(this.loadData, 60000);
    }
    loadData() {
        fetch('internalapi/Transaction/Get?transactionHash=' + this.$route.params.transactionHash)
            .then(response => response.json() as Promise<Transaction>)
            .then(data => {
                this.transaction = data;
            });
    }
    formatCurrency(value: number) {
        let val = (value / 1).toFixed(2).replace(',', '.')
        return val.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",")
    }
}
